src/command/resume_only.rs

```diff
@@ -0,0 +1,131 @@
+use crate::command::into_history_trade;
+use crate::swap::SwapKind;
+use crate::{
+    bitcoin, command::FinishedSwap, config::Settings, ethereum, history::History, swap::Database,
+};
+use chrono::Local;
+use comit::btsieve::{bitcoin::BitcoindConnector, ethereum::Web3Connector};
+use std::sync::{Arc, Mutex};
+
+pub async fn resume_only(
+    settings: Settings,
+    bitcoin_wallet: bitcoin::Wallet,
+    ethereum_wallet: ethereum::Wallet,
+) -> anyhow::Result<()> {
+    let bitcoin_wallet = Arc::new(bitcoin_wallet);
+    let ethereum_wallet = Arc::new(ethereum_wallet);
+
+    #[cfg(not(test))]
+    let db = Arc::new(Database::new(&settings.data.dir.join("database"))?);
+    #[cfg(test)]
+    let db = Arc::new(Database::new_test()?);
+
+    let history = Arc::new(Mutex::new(History::new(
+        settings.data.dir.join("history.csv").as_path(),
+    )?));
+
+    let bitcoin_connector = Arc::new(BitcoindConnector::new(
+        settings.bitcoin.bitcoind.node_url,
+        settings.bitcoin.network,
+    )?);
+    let ethereum_connector = Arc::new(Web3Connector::new(settings.ethereum.node_url));
+
+    respawn_swaps(
+        Arc::clone(&db),
+        Arc::clone(&bitcoin_wallet),
+        Arc::clone(&ethereum_wallet),
+        Arc::clone(&bitcoin_connector),
+        Arc::clone(&ethereum_connector),
+        history,
+    )
+    .await?;
+
+    Ok(())
+}
+
+async fn respawn_swaps(
+    db: Arc<Database>,
+    bitcoin_wallet: Arc<bitcoin::Wallet>,
+    ethereum_wallet: Arc<ethereum::Wallet>,
+    bitcoin_connector: Arc<comit::btsieve::bitcoin::BitcoindConnector>,
+    ethereum_connector: Arc<comit::btsieve::ethereum::Web3Connector>,
+    history: Arc<Mutex<History>>,
+) -> anyhow::Result<()> {
+    for swap in db.all_swaps()?.into_iter() {
+        let finished_swap = execute_swap(
+            Arc::clone(&db),
+            Arc::clone(&bitcoin_wallet),
+            Arc::clone(&ethereum_wallet),
+            Arc::clone(&bitcoin_connector),
+            Arc::clone(&ethereum_connector),
+            swap,
+        )
+        .await?;
+        handle_finished_swap(finished_swap, Arc::clone(&db), Arc::clone(&history))
+    }
```

Comment:
:mega: I think we should be joining all these futures rather than awaiting one by one.

Reply:
Done.